docs/proposals/use-cascading-deletion/README.md

```diff
@@ -0,0 +1,397 @@
+        // +optional
+        Name string `json:"name,omitempty"`
+
+        // A label query over a set of resources.
+        // If name is not empty, labelSelector will be ignored.
+        // +optional
+        LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+}
+
+type CascadeDeletionStatus struct {
+        ...
+}
+```
+
+Work
+```go
+// WorkSpec defines the desired state of Work.
+type WorkSpec struct {
+        // CascadeDeletion Declare the cascade deletion strategy. The default value is null, which is equivalent to background.
+        // +optional
+        CascadeDeletion *CascadeDeletionPolicy `json:"cascadeDeletion,omitempty"`
+
+        ...
+}
+```
+
+#### Controller logic changes
+
+<!--
+binding-controller/cluster-resource-binding-controller 在创建或更新 work 对象的时候, 查询是否存在关联目标资源的 CascadeDeletionPolicy，如果能够找到，将删除策略同步至 Work 对象中。
+execution-controller 根据 Work 对象中的 CascadeDeletion 字段，进行资源删除。
+-->
+
+The `binding-controller`/`cluster-resource-binding-controller` checks for the existence of a `CascadeDeletionPolicy` associated with the target resource when creating or updating the Work object. If found, the deletion policy is synchronized into the Work object.
+
+The `execution-controller` carries out resource deletion based on the `CascadeDeletion` field in the Work object.
+
+#### User usage example
+
+Set the cascade deletion policy to orphan
+
+```yaml
+apiVersion: policy.karmada.io/v1alpha1
+kind: CascadeDeletionPolicy
+metadata:
+  name: foo
+spec:
+  cascadeDeletion: orphan
+  resourceSelectors:
+    - apiVersion: apps/v1
+      kind: Deployment
+      name: foo
+      namespace: default
+```
+
+### Solution Four: Extended by Annotation & Extend the fields of PropagationPolicy/ClusterPropagationPolicy
+
+Equivalent to supporting both solution one and solution two
+
+### Solution comparison
+
+| Name           | Supported control plane resources | Extend API resources | User learning cost | Execute orphan delete just by per resource |
+|----------------|-----------------------------------|----------------------|--------------------|--------------------------------------------|
+| Solution One   | all resources                     | None                 | Lowest             | YES                                        |
+| Solution Two   | resource template                 | PP/CPP/RB/CRB/WORK   | Lowest             | No                                         |
+| Solution Three | all resources                     | new CRD/WORK         | Highest            | YES                                        |
+| Solution Four  | all resources                     | PP/CPP/RB/CRB/WORK   | Lower              | YES                                        |
+
+Solution One:
+Disadvantages:
+- When the `execution-controller` determines whether to cascade delete resources in the member clusters, it needs to parse the resources in the Work template from the manifest, which adds an extra parsing overhead.
```

Comment:
For now, during the process of deleting the workload, the `execution-controller` unmarshals the manifest already, does this approach adds extra overhead?
https://github.com/karmada-io/karmada/blob/b27e669f481a683fa4eb6dec6066e77ccf67a5d2/pkg/controllers/execution/execution_controller.go#L165-L182

Reply:
Good point. For this approach, no extra parsing overhead is required.